pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -0,0 +1,792 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from __future__ import print_function
+from six import StringIO
+from pyutilib.misc import PauseGC
+from pyomo.core.base import (
+    SymbolMap, AlphaNumericTextLabeler, NumericLabeler,
+    Block, Constraint, Expression, Objective, Var, Set, RangeSet, Param,
+    value, minimize, Suffix)
+from pyomo.core.base.component import ComponentData
+import os, sys, subprocess, pipes, math, logging
+
+import pyomo.util.plugin
+from pyomo.opt.base import IOptSolver
+import pyutilib.services
+
+from pyutilib.misc import Options
+
+
+logger = logging.getLogger('pyomo.solvers')
+
+pyutilib.services.register_executable(name="gams")
+ 
+class GAMSSolver(pyomo.util.plugin.Plugin):
+    """A generic interface to GAMS solvers"""
+    pyomo.util.plugin.implements(IOptSolver)
+    pyomo.util.plugin.alias('gams', doc='The GAMS modeling language')
+
+    options = Options()
+
+    def available(self, exception_flag=True):

```

Comment:
This implementation assumes that you are calling the "gams" executable.  You implementation, however, uses the Python API.  Your `available()` implementation therefore needs to return whether the GAMS Python API is importable, not if the gams executable is on the PATH.  In the future, it might be useful to support both the "direct"" (python) interface, and running GAMS through a subprocess.